src/Phinx/Db/Adapter/MysqlAdapter.php

```diff
@@ -98,7 +99,59 @@ public function connect()
             }
         }
     }
-    
+
+    /**
+     * {@inheritdoc}
+     */
+    public function hasSchemaTable()
+    {
+        if (! $this->hasTable($this->getSchemaTableName())) {
+            return false;
+        }
+
+        if (! $this->hasIndex($this->getSchemaTableName(), 'version')) {
+            $this->updateSchemaTable();
+        }
+        return true;
+    }
+
+    protected function updateSchemaTable()
+    {
+        $this->getOutput()->writeln('<info>'.$this->getSchemaTableName() . ' schema out of date. updating schema</info>');
+        try {
+            $oldTable = $this->getSchemaTableName() . '_old';
+            $this->renameTable($this->getSchemaTableName(), $oldTable);
```

Comment:
As we're trying to resolve a potential race condition, these steps aren't going to work. Needs to be one atomic operation, e.g. `ALTER`